seacatauth/credentials/providers/mysql.py

```diff
@@ -0,0 +1,386 @@
+
+
+	async def update(self, credentials_id, update: dict) -> Optional[str]:
+		mysql_id = credentials_id[len(self.Prefix):]
+		updated_fields = list(update.keys())
+
+		assignments = []
+
+		for field, db_field in self.Fields.items():
+			value = update.pop(field, None)
+			if value not in frozenset(["", None]):
+				assignments.append("`{}` = '{}'".format(db_field, value))
+
+		query = "UPDATE `{table}` SET {assignments} WHERE `{id_field}` = {mysql_id};".format(
+			table=self.Table,
+			assignments=", ".join(assignments),
+			id_field=self.IdField,
+			mysql_id=mysql_id,
+		)
+
+		if len(update) != 0:
+			raise KeyError("Some credentials fields cannot be updated: {}".format(", ".join(update.keys())))
+
+		async with aiomysql.connect(**self.ConnectionParams) as connection:
+			async with connection.cursor(aiomysql.DictCursor) as cursor:
+				await cursor.execute(query)
+			try:
+				await connection.commit()
+			except pymysql.err.IntegrityError as e:
+				raise ValueError("Cannot update credentials: {}".format(e)) from e
+
+		L.log(asab.LOG_NOTICE, "Credentials updated", struct_data={
+			"provider_id": self.ProviderID,
+			"cid": credentials_id,
+			"fields": updated_fields
+		})
+		return "OK"
+
+
+	async def delete(self, credentials_id) -> Optional[str]:
+		mysql_id = credentials_id[len(self.Prefix):]
+		query = "DELETE FROM `{table}` WHERE `{field}` = {value};".format(
+			table=self.Table,
+			field=self.IdField,
+			value=mysql_id,
+		)
+		async with aiomysql.connect(**self.ConnectionParams) as connection:
+			async with connection.cursor(aiomysql.DictCursor) as cursor:
+				await cursor.execute(query)
+			try:
+				await connection.commit()
+			except pymysql.err.IntegrityError as e:
+				raise ValueError("Cannot delete credentials: {}".format(e)) from e
+
+		L.log(asab.LOG_NOTICE, "Credentials deleted", struct_data={
+			"provider_id": self.ProviderID,
+			"cid": credentials_id
+		})
+		return "OK"
+
+
+	async def locate(self, ident: str, ident_fields: dict = None) -> Optional[str]:
+		if ident_fields is None:
+			ident_fields = ["username"]
+
+		conditions = []
+		for field, mode in ident_fields.items():
+			field = self.Fields[field]
+			if mode is None:
+				conditions.append("(`{field}` = '{ident}')".format(field=field, ident=ident))
+			if mode == "ignorecase":
+				conditions.append("(LOWER(`{field}`) = '{ident}')".format(field=field, ident=ident))
+		query = "SELECT * FROM `{table}` WHERE {where};".format(
+			table=self.Table,
+			where=" OR ".join(conditions),
+		)
+		async with aiomysql.connect(**self.ConnectionParams) as connection:
+			async with connection.cursor(aiomysql.DictCursor) as cursor:
+				await cursor.execute(query)
+				result = await cursor.fetchone()
+		if result is None:
+			return None
+		return "{}{}".format(self.Prefix, result[self.IdField])
+
+
+	async def get_by(self, key: str, value) -> Optional[dict]:
+		query = "SELECT * FROM `{table}` WHERE `{field}` = {value};".format(
```

Comment:
`SELECT *` is a problematic construct.
You should let people to specify fields to select in the configuration - take advantage of "... AS ..." if needed.